### PR TITLE
docs: update pull request template

### PR DESCRIPTION
**Pull request by codingben**
Signed-off-by: Ben Oukhanov <boukhanov@redhat.com>

**What this PR does / why we need it**:

Cleanup pull request template and add hidden comments for the user.

**Special notes for your reviewer**:

Please see how it looks in the forked repository [here](https://github.com/benukhanov/kubevirt/blob/pull-req-temp/.github/PULL_REQUEST_TEMPLATE.md) (use also raw option).

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
+Add special notes for your reviewer here.
 
 **Release note**:
 <!--  Write your release note:
 1. Enter your extended release note in the below block. If the PR requires additional action from users switching to the new release, include the string "action required".
 2. If no release note is required, just write "NONE".
 -->
 ```release-note
-
+NONE
```
Comment (rmohr) [inline .github/PULL_REQUEST_TEMPLATE.md:20]: I would prefer leaving it empty to force people to think for a moment if a release note should be there.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,19 +1,21 @@
...
+**What are these changes and why are they needed**:
+Describe your changes here.
 
-**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
-Fixes #
+**What issue are you fixing**:
+Fixes #(issue) (Will be closed after merging your changes.)
 
-**Special notes for your reviewer**:
+**Special notes**:
```
Comment (rmohr) [inline .github/PULL_REQUEST_TEMPLATE.md:11]: Would it make sense to align it with kubernetes? https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md

**Review by rmohr — CHANGES_REQUESTED** [review]

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (benukhanov) [reply .github/PULL_REQUEST_TEMPLATE.md]: Sure.

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (benukhanov) [reply .github/PULL_REQUEST_TEMPLATE.md]: Done.

**Comment by benukhanov** [comment]
@rmohr Now you should like it 😄

**Comment by benukhanov** [comment]
@rmohr Please see how it looks [here](https://github.com/benukhanov/kubevirt/blob/pull-req-temp/.github/PULL_REQUEST_TEMPLATE.md) (use also raw option) in the forked repository.

**Comment by codingben** [comment]
Rebased and updated description. @maya-r @dankenigsberg @enp0s3 Can you please re-review it?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,18 +1,35 @@
-<!--  Thanks for sending a pull request!  Here are some tips for you:
```
Comment (dankenigsberg) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: in your commit message, please explain why your suggested template is better than the former one.

Frankly, I'd *add* this request to the template. Every PR should give a reasoning on its motivation.

Comment (codingben) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: I see too much changes in this PR for a single readme file. I'll close this PR and open another in the future.

**Pull request closed without merging**